### PR TITLE
check licenses

### PR DESCRIPTION
**Pull request by romainmenke**
inspired by : https://github.com/csstools/postcss-plugins/pull/475

This action did not work at all but overall I prefer something custom as this is not something that is complex.

- focus on a single ecosystem (npm)
- precisely scan dependencies (we know our own layout)
- precisely scan distributed files (we know our own layout)

-----

> Checks 2 things
> 1. Do all our dependencies have an appropriate license?
> 2. Are there unexpected license comments in build artifacts?
>
> Combined this should give our users sufficient safeguards against license issues.

We are already careful with the dependencies we use and pass on to users and this adds more checks to ensure that we do not accidentally cause issues for the community.

It should give a reasonable guarantee that the combined software our users get is appropriately licensed.

### CODE REVIEW

**Comment by romainmenke** [comment]
check the two failing commits and their annotations for examples of issues and how these are reported.

File: .github/bin/format-package-json.mjs [hunk 1]
```diff
@@ -51,6 +51,9 @@ const formatted = {};
 
 	formatted.license = packageJSONInfo.license;
 	delete packageJSONInfo.license;
+	if (formatted.license !== 'CC0-1.0' && formatted.license !== 'MIT') {
+		formatted.license = 'MIT';
+	}
```
Comment (Antonio-Laguna) [inline .github/bin/format-package-json.mjs:56]: Is this intended?

**Review by Antonio-Laguna — APPROVED** [review]
LGTM! Just a minor question

File: .github/bin/format-package-json.mjs (hunk 1 above)
Comment (romainmenke) [reply .github/bin/format-package-json.mjs:56]: Not really, just copy/paste from the last string on line 54.
I will change to `CC0-1.0` which is the most common license in plugins here.

**Pull request merged**